ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/BeaconStateUtil.java

```diff
@@ -464,6 +464,10 @@ public static UInt64 compute_start_slot_at_epoch(UInt64 epoch) {
     return epoch.times(SLOTS_PER_EPOCH);
   }
 
+  public static boolean is_start_slot_of_epoch(final UInt64 slot) {
+    return slot.mod(SLOTS_PER_EPOCH).equals(UInt64.ZERO);
+  }
```

Comment:
This doesn't appear to be used?

Reply:
at the moment its only in test code but the `RecentChainData.updateChainHead` will call it so i dont have to code the decision into that...